packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -1315,6 +1315,28 @@ void main() {
     expect(tester.takeException(), isNull);
   });
 
+  // Regression test for https://github.com/flutter/flutter/issues/154532.
+  testWidgets('keyboard navigation does not throw when entries are filtered empty',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+          body: DropdownMenu<TestMenu>(
+            enableFilter: true,
+            dropdownMenuEntries: menuChildren,
+          ),
+        ),
+      ));
+      await tester.tap(find.byType(DropdownMenu<TestMenu>));
+      await tester.pump();
+      await tester.enterText(find.byType(TextField).first, 'No match');
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.arrowDown);
+      await tester.pump();
+      await tester.enterText(find.byType(TextField).first, 'No match 2');
+      await tester.pump();
+      expect(tester.takeException(), isNull);
+  }, variant: TargetPlatformVariant.desktop());
```

Comment:
Keyboard navigation is also supported on mobile (Android for sure, iOS probably).
I tried changing `TargetPlatformVariant.desktop()` to `TargetPlatformVariant.all()` and the test still passed.

Reply:
The different is `requestFocusOnTap` default to false on mobile, The widget won't gain focus and won't receive key event. (I think this might be the reason other keyboard navigation tests used `TargetPlatformVariant.desktop()` )

I can see setting `requestFocusOnTap` to `true` being overall the better solution than setting `TargetPlatformVariant`

Reply:
Thanks for the explanation. :pray: 
For the moment, it makes sense to do the same as the other existing tests.
We can revisit this on a separate "test only" PR (feel free to ping me if you file one, I like PRs that aim at improving existing tests :fire: ), setting `requestFocusOnTap` to `true` seems to be a nice option (a comment might be interesting to remind why).

Reply:
Oops sorry, I was reading the wrong version. I see now that you applies this change, it is ok to me. Feel free to add a comment if you think it worths it.

Reply:
Ok, I've changed it back to `TargetPlatformVariant.desktop()` for now.

Reply:
Oops, it appears to be some delays and I didn't see the new message, I will try to file a PR about overall `requestFocusOnTap` in keyboard navigation in a few days.